src/Functions/FunctionsJSON.cpp

```diff
@@ -1332,13 +1390,64 @@ class JSONExtractRawImpl
 
     static bool insertResultToColumn(IColumn & dest, const Element & element, std::string_view)
     {
-        ColumnString & col_str = assert_cast<ColumnString &>(dest);
-        auto & chars = col_str.getChars();
-        WriteBufferFromVector<ColumnString::Chars> buf(chars, AppendModeTag());
+        if (dest.getDataType() == TypeIndex::LowCardinality)
+        {
+            ColumnString::Chars chars;
+            WriteBufferFromVector<ColumnString::Chars> buf(chars, AppendModeTag());
+            traverse(element, buf);
+            buf.finalize();
+            assert_cast<ColumnLowCardinality &>(dest).insertData(reinterpret_cast<const char *>(chars.data()), chars.size());
```

Comment:
If I understand correctly, in this case `dest` is `LowCardinality(String)` and we need to add `\0` at the end of a string

Reply:
The way I understood it, `ColumnLowCardinality` does not need a `\0` ant the end, only `ColumnString`.
Nonetheless, I tried to add a `\0` with `chars.push_back(0)` at the end in this case, and I am getting `{"b":{"c":1,"d":"str"}}\0` from executing `SELECT JSONExtract('{"a" : {"b" : {"c" : 1, "d" : "str"}}}', 'Tuple( a LowCardinality(String))');`, which looks wrong and is not happening without the `chars.push_back(0)`